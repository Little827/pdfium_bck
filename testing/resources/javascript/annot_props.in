{{header}}
{{object 1 0}} <<
  /Type /Catalog
  /Pages 2 0 R
  /OpenAction 20 0 R
>>
endobj
{{object 2 0}} <<
  /Type /Pages
  /Count 1
  /Kids [10 0 R]
>>
endobj
{{object 10 0}} <<
  /Type /Page
  /Parent 2 0 R
  /Resources <<
    /Font <</F1 15 0 R>>
  >>
  /MediaBox [0 0 612 792]
  /Annots [22 0 R]
  /Tabs /R
>>
endobj
{{object 20 0}} <<
  /Type /Action
  /S /JavaScript
  /JS 21 0 R
>>
endobj
{{object 21 0}} <<
>>
stream
var annot = this.getAnnot(0, "Annot-1");
try {
  app.alert("hidden: " + annot.hidden);
  app.alert("name: " + annot.name);
  app.alert("type: " + annot.type);
  annot.hidden = true;
  annot.name = "clams";
  app.alert("hidden: " + annot.hidden);
  app.alert("name: " + annot.name);
} catch (e) {
  app.alert("ERROR: " + e);
}
try {
  annot.type = 42;
} catch (e) {
  app.alert("SUCCESS: " + e);
}
endstream
endobj
{{object 22 0}} <<
  /Type /Annot
  /Subtype /Highlight
  /Rect [475 681 512 690]
  /NM (Annot-1)
  /F 2
  /QuadPoints [475 688 512 688 475 679 512 679]
  /C [0.0001108646 0.001760244 0.9982184]
  /Contents ()
>>
endobj
{{xref}}
{{trailer}}
{{startxref}}
%%EOF
