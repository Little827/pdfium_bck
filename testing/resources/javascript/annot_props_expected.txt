Alert: hidden: true
Alert: name: Annot-1
Alert: type: Highlight
Alert: hidden now: true
Alert: hidden now: false
Alert: name now: clams
Alert: SUCCESS: Annot.type: Cannot assign to readonly property.
